### ISSUE TITLE
Replace dict.has_key() with the 'in' keyword for twisted/names, twisted/news, twisted/persisted, twisted/protocols, twisted/runner

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8363]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8363|
|Type|enhancement|
|Created|2016-05-26 17:51:20Z|
|Branch|https://github.com/twisted/twisted/tree/8363-rodrigc-has_key5|

In PEP 0290, has_key() was deprecated in Python 2.2 and higher: 
* https://www.python.org/dev/peps/pep-0290/#testing-dictionary-membership
* https://docs.python.org/2.2/whatsnew/node4.html

In Python 3, dict.has_key() was removed:
* https://docs.python.org/3.0/whatsnew/3.0.html#builtins

PEP 0290 says that for testing dictionary membership, this:

```
if d.has_key(k):
```

should be changed to this:

```
if k in d:
```

<details><summary>Searchable metadata</summary>

```
trac-id__8363 8363
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8363_rodrigc_has_key5 8363-rodrigc-has_key5
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464285080613047 1464285080613047
changetime__1465458143281925 1465458143281925
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/79

This was pulled out from [ticket:4053]

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Looks good, thanks!
I have also created a topfile.

Will merge after buildbot is done.

thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 08d0935768767ee95ead2a21ec0c07efa844ff5d
```
#!CommitTicketReference repository="" revision="08d0935768767ee95ead2a21ec0c07efa844ff5d"
Merge 8363-rodrigc-has_key5: Replace dict.has_key() in various places.

Replace dict.has_key() with the 'in' keyword for:
* twisted/names
* twisted/news
* twisted/persisted
* twisted/protocols
* twisted/runner

Author: rodrigc
Reviewer: adiroiban
Fixes: #8363
```
